android-components/components/browser/icons/src/main/java/mozilla/components/browser/icons/decoder/SvgIconDecoder.kt

```diff
@@ -65,6 +65,8 @@ class SvgIconDecoder(val context: Context) : ImageDecoder {
             svg.renderToCanvas(Canvas(bitmap))
 
             bitmap
+        } catch (e: IllegalArgumentException) {
+            null
```

Comment:
It might be good to also add a logger error here

Reply:
Instead of catching this, could we not pass the illegal argument in the affected function call?  That way we handle the edge case properly and prevent throwing an exception to catch it.

Reply:
Coik-kt use https://github.com/coil-kt/coil/blob/e0411dd23c40dc05d69b8d05cb19b725f2d83a36/coil-svg/src/main/java/coil/decode/DecodeUtils.kt#L18-L20 to try to ensure it is a valid xml (but with no guarantee) 

https://github.com/coil-kt/coil/blob/e0411dd23c40dc05d69b8d05cb19b725f2d83a36/coil-svg/src/main/java/coil/decode/SvgDecoder.kt#L104

Reply:
It is so sad that kotlin don't have checked exceptions :(